tools/python/paladin_claiming/recycle_bribes.py

```diff
@@ -0,0 +1,196 @@
+    response = requests.get(
+        f"https://api.paladin.vote/quest/v3/copilot/claims/{address}"
+    )
+    response.raise_for_status()
+    return response.json()["claims"]
+
+
+def get_prop_hash(platform: str, target: str) -> str:
+    if platform == "bal":
+        prop = Web3.solidity_keccak(["address"], [Web3.to_checksum_address(target)])
+        return f"0x{prop.hex().lstrip('0x')}"
+    if platform == "aura":
+        return get_hh_aura_target(target)
+    raise ValueError(f"platform {platform} not supported")
+
+
+def claim_bribes(claims: List[dict], chain_name: str) -> str:
+    """
+    - generate tx to claim all bribes from paladin
+    - save claim results to csv
+    """
+    chain_addrs = CHAIN_ADDRS[chain_name]
+    paladin_distributor = SafeContract(
+        chain_addrs["paladin_distributor"], paladin_distributor_abi
+    )
+    w3 = w3_by_chain[chain_name]
+    csv_path = CLAIM_OUTPUT_DIR / f"paladin_claims_{chain_name}.csv"
+
+    with open(csv_path, "w") as f:
+        f.write("chain,source,token_address,gauge_address,amount\n")
+
+        for claim in claims:
+            print(f"Claiming - token: {claim['token']}, gauge: {claim['gauge']}")
+
+            paladin_distributor.claim(
+                claim["questId"],
+                claim["period"],
+                claim["index"],
+                claim["user"],
+                claim["amount"],
+                claim["proofs"],
+            )
+
+            token_contract = w3.eth.contract(
+                address=Web3.to_checksum_address(claim["token"]), abi=erc20_abi
+            )
+            decimals = token_contract.functions.decimals().call()
+            human_amount = int(claim["amount"]) / (10**decimals)
+
+            f.write(
+                f"{chain_name},{claim['path']},{claim['token']},{claim['gauge']},{human_amount}\n"
+            )
+
+    return csv_path
+
+
+def deposit_bribes(csv_path: str, chain_name: str, builder: SafeTxBuilder):
+    """
+    process claimed bribes from csv and deposit to HH
+    - generate tx to deposit 70% of claimed bribe amounts
+    - save remaining 30% (DAO fee) to separate csv
+    - output final transaction payload for claims and deposits
+    """
+    chain_addrs = CHAIN_ADDRS[chain_name]
+
+    sell_csv_path = CLAIM_OUTPUT_DIR / f"paladin_tokens_to_sell_{chain_name}.csv"
+    with open(sell_csv_path, "w") as f:
+        f.write("chain,token_address,amount_wei,amount_human,decimals\n")
+
+    bribe_data = {}
+    if chain_name == "arbitrum":
+        # only aura briber on arbitrum
+        aura_briber = SafeContract(chain_addrs["aura_briber"], bribe_market_abi)
+        bribe_data["bal"] = (aura_briber, 1)
+        bribe_data["aura"] = (aura_briber, 1)
+    else:
+        bal_briber = SafeContract(chain_addrs["bal_briber"], bribe_market_abi)
+        bribe_data["bal"] = (bal_briber, 2)
+
+        aura_briber = SafeContract(chain_addrs["aura_briber"], bribe_market_abi)
```

Comment:
no aura bribs, we're trying to not actually, literally recycle :). If we move them to a new market we will never earn them again.

Reply:
fixed in [0740502](https://github.com/BalancerMaxis/multisig-ops/pull/1488/commits/0740502d06a3cfa22535e1490603f097d1711cd5)

Reply:
@jalbrekt85 @Tritium-VLK is maybe also an idea to blacklist these two or three gho and rpl markets to even prevent the need for recycling?

Reply:
> @jalbrekt85 @Tritium-VLK is it maybe also an idea to blacklist these two or three gho and rpl markets to even prevent the need for recycling?

Blacklist how?  These are hidden hand bribs.  We are already blacklisted if that is what you mean.

Reply:
That would limit the amount of pools BD can incentivise.  This may make sense, there have been and likely will be technical reasons why it won't always work hence this script.